### ISSUE TITLE
Rendering 3000 instances of a box reduces framerate to 14fps

### ISSUE THREAD

**Issue by bnolan** [BabylonJS/BabylonNative#205]
I got things running, works really nicely on Os X, performance only 14fps for rendering 3000 boxes using [this gist](https://gist.github.com/bnolan/c9426f115c6b3b4fb9c2dde7455d3e32). Running on a macbook pro 15" with AMD Radeon Pro 455.

Thread 6 maxed out:

<img width="461" alt="Screen Shot 2020-04-06 at 8 21 19 PM" src="https://user-images.githubusercontent.com/17499/78537860-3f1a0300-7844-11ea-9686-2232cade27b0.png">

Paused in thread 6:

<img width="845" alt="Screen Shot 2020-04-06 at 8 21 34 PM" src="https://user-images.githubusercontent.com/17499/78537882-45a87a80-7844-11ea-885c-d2ecccd5a965.png">

**Comment by bnolan** [comment]
I'm not sure I'm reading this right - but it looks like the GPU takes a ~450us to render the frame, but there is ~110 ms of idle time for the frame - which would match up with the ~14fps reading I'm getting.

<img width="1297" alt="Screen Shot 2020-04-06 at 8 28 14 PM" src="https://user-images.githubusercontent.com/17499/78538470-28c07700-7845-11ea-8f4c-16a0a05c588b.png">

Metal seems to show ~100ms per frame.

<img width="1059" alt="Screen Shot 2020-04-06 at 8 29 38 PM" src="https://user-images.githubusercontent.com/17499/78538608-5e656000-7845-11ea-8be8-656f1aca83bc.png">

Sample frame:

<img width="1180" alt="Screen Shot 2020-04-06 at 8 02 13 PM" src="https://user-images.githubusercontent.com/17499/78538635-67563180-7845-11ea-9bc2-33ee7de2a919.png">

**Comment by CedricGuillemet** [comment]
I'm taking a look!

**Comment by CedricGuillemet** [comment]
To know if it's related to Metal/JSC or something else, I profiled your PG on Windows using Chakra and D3D11. I get the same result: around 12FPS on my work PC. Most of the time is spent in updating the uniforms.
![image](https://user-images.githubusercontent.com/1312968/78783481-fc6f3c80-79a3-11ea-81ce-4571e7cc85fb.png)
napi counts for <9% of time.

**Comment by CedricGuillemet** [comment]
A while ago, I did a PR on bgfx to have finer granularity on state preservation but I didn't take time to update it for BabylonNative. 
https://github.com/bkaradzic/bgfx/pull/2067
I've tested it today and the result is pretty great.
As uniform are reset for every drawcall, I discard the preservation for bgfx::submit.
```
bgfx::submit(m_frameBufferManager.GetBound().ViewId, m_currentProgram->Program, 0, BGFX_DISCARD_INSTANCE_DATA|BGFX_DISCARD_STATE|BGFX_DISCARD_TRANSFORM);
```
And I'm back at 60FPS! It'll be pushed in master in the coming days. Time for me to test other platforms.
rendererUpdateUniforms is gone!
![image](https://user-images.githubusercontent.com/1312968/78791465-31818c00-79b0-11ea-886d-56423bbf575f.png)
